src/instruments/src/SD/Pages/Apu/Apu.tsx

```diff
@@ -0,0 +1,521 @@
+
+import './Apu.scss';
+import { EcamPage } from '../../Common/EcamPage';
+import { SvgGroup } from '../../Common/SvgGroup';
+
+setIsEcamPage('apu_page');
+
+export const ApuPage = () => {
+    const [apuAvail] = useSimVar('L:A32NX_OVHD_APU_START_PB_IS_AVAILABLE', 'Bool', 1000);
+
+    return (
+        <EcamPage name="main-apu">
+            <PageTitle x={283} y={33} text="APU" />
+
+            {/* APU Avail */}
+            {apuAvail
+            && (
+                <SvgGroup x={305} y={85}>
+                    <text x={0} y={0} className="Green FontTitle Center">AVAIL</text>
+                </SvgGroup>
+            )}
+
+            <ApuGen x={105} y={100} />
+
+            <ApuBleed x={420} y={153} />
+
+            {/* Separation Bar */}
+            <SvgGroup x={83} y={252}>
+                <line className="Line Grey" x1={0} y1={0} x2={0} y2={26} />
+                <line className="Line Grey" x1={-1} y1={0} x2={456} y2={0} />
+                <line className="Line Grey" x1={455} y1={0} x2={455} y2={26} />
+            </SvgGroup>
+
+            <NGauge x={155} y={295} />
+
+            <EgtGauge x={155} y={410} />
+
+            <ApuMemos x={370} y={335} />
+        </EcamPage>
+    );
+};
+
+type ComponentPositionProps = {
+    x: number,
+    y: number,
+}
+
+const ApuGen = ({ x, y } : ComponentPositionProps) => {
+    const [apuContactorClosed] = useSimVar('L:A32NX_ELEC_CONTACTOR_3XS_IS_CLOSED', 'Bool', 1000);
+    const [busTieContactor1Closed] = useSimVar('L:A32NX_ELEC_CONTACTOR_11XU1_IS_CLOSED', 'Bool');
+    const [busTieContactor2Closed] = useSimVar('L:A32NX_ELEC_CONTACTOR_11XU2_IS_CLOSED', 'Bool');
+
+    const [apuGenLoad] = useSimVar('L:A32NX_ELEC_APU_GEN_1_LOAD', 'Percent', 500);
+    const [apuGenLoadNormalRange] = useSimVar('L:A32NX_ELEC_APU_GEN_1_LOAD_NORMAL', 'Bool', 750);
+
+    const [apuGenVoltage] = useSimVar('L:A32NX_ELEC_APU_GEN_1_POTENTIAL', 'Volts', 500);
+    const [apuGenPotentialNormalRange] = useSimVar('L:A32NX_ELEC_APU_GEN_1_POTENTIAL_NORMAL', 'Bool', 750);
+
+    const [apuGenFreq] = useSimVar('L:A32NX_ELEC_APU_GEN_1_FREQUENCY', 'Hertz', 500);
+    const [apuGenFreqNormalRange] = useSimVar('L:A32NX_ELEC_APU_GEN_1_FREQUENCY_NORMAL', 'Bool', 750);
+
+    enum apuGenState {STANDBY, OFF, ON}
+    const [currentApuGenState, setCurrentApuGenState] = useState(apuGenState.STANDBY);
+
+    const [apuMasterPbOn] = useSimVar('L:A32NX_OVHD_APU_MASTER_SW_PB_IS_ON', 'Bool', 500);
+    const [apuGenPbOn] = useSimVar('A:APU GENERATOR SWITCH', 'Boolean', 750);
+    const [apuAvail] = useSimVar('L:A32NX_OVHD_APU_START_PB_IS_AVAILABLE', 'Bool', 1000);
+
+    useEffect(() => {
+        if (!apuMasterPbOn && !apuAvail /* FIXME: and apu master switch is valid and apu gen push buttion is valid */) {
+            setCurrentApuGenState(apuGenState.STANDBY);
+        } else if (!(currentApuGenState === apuGenState.STANDBY) && !apuGenPbOn /* FIXME: and apu master switch valid and apu gen pb valid */) {
```

Comment:
**thought**: I would remove these `FIXME` comments. As they're describing a level of detail we're not close to achieving any time soon. These comments are saying: if we want to have an even higher level of detail we need to do something here. The problem is that such comments should probably go on every other line in our code base. Let's not do that 😀.

Reply:
**suggestion**: Use `!==` instead of `!(===)`
```suggestion
        if (!apuMasterPbOn && !apuAvail /* FIXME: and apu master switch is valid and apu gen push buttion is valid */) {
            setCurrentApuGenState(apuGenState.STANDBY);
        } else if (currentApuGenState !== apuGenState.STANDBY && !apuGenPbOn /* FIXME: and apu master switch valid and apu gen pb valid */) {
```

Reply:
**suggestion**: Consider replacing the `useEffect` with the code below. Note that this doesn't necessarily fix the bug, but might make things easier to reason about.

```typescript
const isStandby = !apuMasterPbOn && !apuAvail;
const isOff = !isStandby && !apuGenPbOn;
const isOn = !isStandby && !isOff;
```

Reply:
> **thought**: I would remove these `FIXME` comments. As they're describing a level of detail we're not close to achieving any time soon. These comments are saying: if we want to have an even higher level of detail we need to do something here. The problem is that such comments should probably go on every other line in our code base. Let's not do that 😀.

You raise a very good point. I have gotten rid of all of the todo’s and left a couple FBWDOC comments in the components for us to reference later on when we have the means to implement what we currently can’t. :)